pennylane/collections/dot.py

```diff
@@ -124,7 +124,10 @@ def dot(x, y):
     >>> cost(x)
     tensor(-0.2183, dtype=torch.float64, grad_fn=<DotBackward>)
     """
-    warnings.warn("The dot function is deprecated and it will be removed soon.", UserWarning)
+    warnings.warn(
+        "The dot function of the collections module is deprecated and it will be removed soon.",
+        UserWarning,
+    )
```

Comment:
A lot of the changed files are still using `UserWarning`s instead of `DeprecationWarning`. There are other files apart from this one that need to be changed as well, and the tests that check for warnings need to be updated accordingly.

```suggestion
    warnings.warn(
        "The dot function of the collections module is deprecated and it will be removed soon.",
        DeprecationWarning,
    )
```